diff --git a/fingerprintd/FingerprintDaemonProxy.cpp b/fingerprintd/FingerprintDaemonProxy.cpp
index 1c7da30c3..7a4d31866 100644
--- a/fingerprintd/FingerprintDaemonProxy.cpp
+++ b/fingerprintd/FingerprintDaemonProxy.cpp
@@ -48,6 +48,7 @@ void FingerprintDaemonProxy::hal_notify_callback(const fingerprint_msg_t *msg) {
         ALOGE("Invalid callback object");
         return;
     }
+
     const int64_t device = (int64_t) instance->mDevice;
     switch (msg->type) {
         case FINGERPRINT_ERROR:
@@ -74,11 +75,11 @@ void FingerprintDaemonProxy::hal_notify_callback(const fingerprint_msg_t *msg) {
             ALOGD("onEnrollResult(fid=%d, gid=%d, rem=%d)",
                     msg->data.enroll.finger.fid,
                     msg->data.enroll.finger.gid,
-                    msg->data.enroll.samples_remaining);
+                    100 - msg->data.enroll.samples_remaining);
             callback->onEnrollResult(device,
                     msg->data.enroll.finger.fid,
                     msg->data.enroll.finger.gid,
-                    msg->data.enroll.samples_remaining);
+                    100 - msg->data.enroll.samples_remaining);
             break;
         case FINGERPRINT_TEMPLATE_REMOVED:
             ALOGD("onRemove(fid=%d, gid=%d)",
